x-pack/plugins/xpack_main/public/components/telemetry/telemetry_form.js

```diff
@@ -159,25 +158,9 @@ export class TelemetryForm extends Component {
     </Fragment>
   )
 
-  toggleOptIn = async () => {
+  toggleOptIn = () => {
     const newOptInValue = !this.props.telemetryOptInProvider.getOptIn();
-
-    return new Promise((resolve, reject) => {
-      this.setState({
-        enabled: newOptInValue,
```

Comment:
I pulled down locally and gave it a try, it seems the save button no longer hides after you click it.

From what I see, the state use to update after a successful save, which caused the field to re-evaluate `value: telemetryOptInProvider.getOptIn() || false`. We could set a default prop value to `enabled` using `telemetryOptInProvider.getOptIn() || false` and make the field `value: this.props.enabled` which would cause the save button to hide as soon as `enabled` value is toggled in the state.